packages/p2p/src/components/composite-calendar/composite-calendar.tsx

```diff
@@ -0,0 +1,210 @@
+    const { from, to, onChange } = props;
+
+    const [show_to, setShowTo] = React.useState(false);
+    const [show_from, setShowFrom] = React.useState(false);
+    const [days_duration_list] = React.useState([
+        {
+            value: 'all_time',
+            label: localize('All time'),
+            onClick: () => selectDateRange(),
+            duration: 0,
+        },
+        {
+            value: 'today',
+            label: localize('Today'),
+            onClick: () => selectDateRange(1),
+            duration: 1,
+        },
+        {
+            value: 'last_7_days',
+            label: localize('Last 7 days'),
+            onClick: () => selectDateRange(7),
+            duration: 7,
+        },
+        {
+            value: 'last_30_days',
+            label: localize('Last 30 days'),
+            onClick: () => selectDateRange(30),
+            duration: 30,
+        },
+        {
+            value: 'last_60_days',
+            label: localize('Last 60 days'),
+            onClick: () => selectDateRange(60),
+            duration: 60,
+        },
+        {
+            value: 'last_quarter',
+            label: localize('Last quarter'),
+            onClick: () => selectDateRange(90),
+            duration: 90,
+        },
+    ]);
+
+    const wrapper_ref = React.useRef<HTMLInputElement>(null);
+
+    const validateClickOutside = (event: MouseEvent) => !wrapper_ref.current?.contains(event.target as Node);
+
+    const selectDateRange = (new_from?: number) => {
+        hideCalendar();
+        onChange({
+            from: new_from ? toMoment().startOf('day').subtract(new_from, 'day').add(1, 's') : undefined,
+            to: toMoment().endOf('day'),
+            is_batch: true,
+        });
+    };
+
+    const getToDateLabel = () => {
+        const date = toMoment(to);
+        return daysFromTodayTo(date) === 0 ? localize('Today') : date.format('MMM, DD YYYY');
+    };
+
+    const getFromDateLabel = () => {
+        const date = toMoment(from);
+        return from ? date.format('MMM, DD YYYY') : '';
+    };
+
+    const hideCalendar = () => {
+        setShowFrom(false);
+        setShowTo(false);
+    };
+
+    const showCalendar = (label: string) => {
+        if (label === 'from') {
+            setShowFrom(true);
+        }
+        if (label === 'to') {
+            setShowTo(true);
+        }
+    };
+
+    useOnClickOutside(
+        wrapper_ref,
+        (event: MouseEvent) => {
+            event?.stopPropagation();
+            event?.preventDefault();
+            hideCalendar();
+        },
+        validateClickOutside
+    );
+
+    const setFromToDate = (date: moment.Moment, label: string) => {
+        if (label === 'to') {
+            onChange({ to: toMoment(date).endOf('day') });
+        } else {
+            onChange({ from: toMoment(date) });
+            hideCalendar();
+        }
+    };
+
+    const isPeriodDisabled = (date: moment.Moment, label: string) => {
```

Comment:
since its a function, better to name it like this, since the convention for is variables should be like this `is_period_disabled`

```suggestion
    const getIsPeriodDisabled = (date: moment.Moment, label: string) => {
```